frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -795,24 +806,34 @@ export const ClusterServiceVersionsDetailsPage: React.FC<ClusterServiceVersionsD
   props,
 ) => {
   const instancePagesFor = (obj: ClusterServiceVersionKind) => {
+    const internalObjects = getInternalObjects(obj);
     return (providedAPIsFor(obj).length > 1
       ? [{ href: 'instances', name: 'All Instances', component: ProvidedAPIsPage }]
       : ([] as Page[])
     ).concat(
-      providedAPIsFor(obj).map((desc: CRDDescription) => ({
-        href: referenceForProvidedAPI(desc),
-        name: desc.displayName,
-        component: React.memo(
-          () => (
-            <ProvidedAPIPage
-              csv={obj}
-              kind={referenceForProvidedAPI(desc)}
-              namespace={obj.metadata.namespace}
-            />
-          ),
-          _.isEqual,
-        ),
-      })),
+      providedAPIsFor(obj).reduce(
+        (acc, desc: CRDDescription) =>
+          !isInternalObject(internalObjects, desc.name)
+            ? [
+                {
+                  href: referenceForProvidedAPI(desc),
+                  name: desc.displayName,
+                  component: React.memo(
+                    () => (
+                      <ProvidedAPIPage
+                        csv={obj}
+                        kind={referenceForProvidedAPI(desc)}
+                        namespace={obj.metadata.namespace}
+                      />
+                    ),
+                    _.isEqual,
```

Comment:
Just wondering, why are we customizing the default `propsAreEqual` comparator when using `React.memo` here?

Reply:
After going through the article you just sent. React does shallowEquals by default. The original author for some reason wanted to do a deep comparison.